### DISCUSSION TITLE
Maximum sampling rate

### DISCUSSION THREAD

**Discussion by MontyJava** [David00/rpi-power-monitor#130] [General]
Hello, I am an electrician interested in an alternative to the Fluke for power logging in order to assess issues related to power quality and malfunctioning devices. Often, I am looking for sudden spikes that can occur well under 1sec in duration. I see that your project has a minimum of 1sec samples, and given the Pi can operate at a much higher frequency, I want to know if this rate can be adjusted higher or was settled due to a hardware limitation.
Thank you.

**Comment by David00** [comment]
Hi @MontyJava, this is an interesting question.  The power monitor actually does collect _individual_ samples quite fast - up to 35,000 times per second, aka: 35 kSPS.  When this bandwidth is shared between the 6 channels, it's effectively about 3 kSPS per channel, including the AC voltage input.  

The 1 second interval you're referring to is the effective window size for a single measurement/datapoint that gets stored in the database.  It's basically like - take all ~3k samples from each channel over a 1 second period, and find the RMS amperage, RMS voltage, and real power from the perspective of each of the six current transformers.

It's a fine balance between chewing up disk space (and chewing up microSD cards!) with constant writes, but still having enough "responsiveness" to feel like it's providing real time data.

It certainly can run much faster, especially if the data is not needing to be written to the disk.

What kind of spikes are you interested in detecting? (I've thought about this before, specifically for inrush current; it'd be neat to have a historical record of inrush currents for my AC condenser unit motors!)

**Reply by MontyJava** [reply]
Thank you for your response.
I am interested in small spikes and the like caused my inrush or SMPS or line side power quality issues that can cause problems for various electrical devices such as GFI AFI and various LEDs generally. Spikes that can happen in microsecond or faster time frames.
